Azure.Data.Wrappers/StorageQueue.cs

```diff
@@ -177,19 +195,30 @@ public virtual CloudQueue Reference
         /// <returns>Task</returns>
         public virtual async Task Send(object obj)
         {
-            if (null == obj)
+            await this.SendAsync(obj);
+        }
+
+        /// <summary>
+        /// Save Model to queue, as json
+        /// </summary>
+        /// <param name="message">object</param>
+        /// <returns>Task</returns>
+        public virtual async Task SendAsync<T>(T message)
+        {
+            if (null == message)
             {
                 throw new ArgumentNullException("obj");
```

Comment:
throw new ArgumentNullException("message");

Reply:
I would prefer not to pull down the stack. Messages coming off of the bus should not be null.